tests/conftest.py

```diff
@@ -204,42 +200,125 @@ def labwhere_samples_error(app, mocked_responses):
 
     body = json.dumps([])
     mocked_responses.add(
-        responses.POST,
-        labwhere_url,
-        body=body,
-        status=HTTPStatus.INTERNAL_SERVER_ERROR,
+        responses.POST, labwhere_url, body=body, status=HTTPStatus.INTERNAL_SERVER_ERROR,
     )
 
 
 @pytest.fixture
-def mlwh_lh_samples(app, sql_engine):
-    insert_lh_samples_into_mlwh(app, MLWH_SEED_SAMPLES, sql_engine)
+def samples_for_mlwh_update(cog_uk_ids):
+    return SAMPLES_FOR_MLWH_UPDATE
 
 
 @pytest.fixture
-def mlwh_lh_samples_multiple(app, sql_engine):
-    insert_lh_samples_into_mlwh(app, MLWH_SEED_SAMPLES_MULTIPLE, sql_engine)
+def cog_uk_ids():
+    return COG_UK_IDS
+
 
+# ********************** WAREHOUSE DATA ************************** #
+
+
+@pytest.fixture
+def mlwh_lh_samples(app, mlwh_sql_engine):
+    insert_into_mlwh(
+        app, MLWH_LH_SAMPLES, mlwh_sql_engine, app.config["MLWH_LIGHTHOUSE_SAMPLE_TABLE"]
+    )
 
-def insert_lh_samples_into_mlwh(app, samples, sql_engine):
-    table = get_table(sql_engine, app.config["MLWH_LIGHTHOUSE_SAMPLE_TABLE"])
 
-    with sql_engine.begin() as connection:
+@pytest.fixture
+def mlwh_lh_samples_multiple(app, mlwh_sql_engine):
+    insert_into_mlwh(
+        app, MLWH_LH_SAMPLES_MULTIPLE, mlwh_sql_engine, app.config["MLWH_LIGHTHOUSE_SAMPLE_TABLE"]
+    )
+
+
+@pytest.fixture
+def mlwh_sample_stock_resource(app, mlwh_sql_engine):
+    # deletes
+    delete_from_mlwh(
```

Comment:
Rather than needing to delete records from various tables when setting up features, it might be nice to adjust:
```
@pytest.fixture
def mlwh_sql_engine(app):
    return create_mysql_connection_engine(
        app.config["WAREHOUSES_RW_CONN_STRING"], app.config["ML_WH_DB"]
    )


@pytest.fixture
def event_wh_sql_engine(app):
    return create_mysql_connection_engine(
        app.config["WAREHOUSES_RW_CONN_STRING"], app.config["EVENTS_WH_DB"]
    )
```
To use a context manager and wrap each test in a transaction. This would decouple the tests, and if my Ruby experience is transferable, should improve test performance.

Bit on sessions and transactions:
https://docs.sqlalchemy.org/en/13/orm/session_transaction.html
Context managers:
https://jeffknupp.com/blog/2016/03/07/python-with-context-managers/

Crawler achieves something similar via truncation:
https://github.com/sanger/crawler/blob/ca6eeefbee2753bd4f93b39c4c00ff5e122930fa/tests/conftest.py#L60

Reply:
As discussed, will look at this as a follow up release.